e2e/tests/functional/plugins/gauge/gauge.e2e.spec.js

```diff
@@ -133,4 +133,50 @@ test.describe('Gauge', () => {
 
     // TODO: Verify changes in the UI
   });
+
+  test('Gauge does not display NaN when data not available', async ({ page }) => {
+    // Create a Gauge
+    await createDomainObjectWithDefaults(page, {
+      type: 'Gauge',
+      name: 'Unnamed Gauge'
+    });
+
+    // Create a Sine Wave Generator in the Gauge with a loading delay
+    await page.getByRole('button', { name: ' Create ' }).click();
+    await page.getByRole('menuitem', { name: ' Sine Wave Generator' }).click();
+    await page
+      .getByRole('dialog')
+      .locator('div')
+      .filter({ hasText: 'Loading Delay (ms)' })
+      .nth(3)
+      .click();
+    await page.getByLabel('Loading Delay (ms)').fill('5000');
+    await page.locator('[aria-label="Save"]').click();
+    await page.getByLabel('Expand My Items folder').click();
+    await page
+      .getByRole('treeitem', { name: 'Expand Unnamed Gauge gauge  Unnamed Gauge' })
+      .locator('a')
+      .click();
+    const gaugeNoDataText = await page.locator('.js-dial-current-value tspan').textContent();
+    expect(gaugeNoDataText).toBe('--');
+  });
+
+  test('Gauge enforces composition policy', async ({ page }) => {
+    // Create a Gauge
+    await createDomainObjectWithDefaults(page, {
+      type: 'Gauge',
+      name: 'Unnamed Gauge'
+    });
+
```

Comment:
We should add a positive case here. e.g.:
1. Create an allowed object into the Gauge
2. Assert that the Save button is enabled

Reply:
So I'm trying to figure out what the scope of the test that I add should be.... gauge.e2e 'Can add and remove telemetry sources' at line 40 already adds a SWG at line 47, and exercises the Save button at line 58. Do I need to duplicate that?

Reply:
Ah, in that case I suppose that's already covered then.